### ISSUE TITLE
Create controlplanerevisions timed out during several int tests 

### ISSUE THREAD

**Issue by atul7799-dev** [terraform-google-modules/terraform-google-kubernetes-engine#2015]
### TL;DR

During several recent tests , the create_cpr.sh is timing out after 10m+ during the asm module deployment

### Expected behavior

should be timeout 

### Observed behavior

 Error: local-exec provisioner error
│ 
│   with module.asm[0].module.cpr.module.gcloud_kubectl.null_resource.run_command[0],
│   on .terraform/modules/asm.cpr/main.tf line 232, in resource "null_resource" "run_command":
│  232:   provisioner "local-exec" {

+ .terraform/modules/asm/modules/asm/scripts/create_cpr.sh
│ asm-managed-stable stable true false
│ Ensuring ControlPlaneRevision exists in cluster... attempt 1
│ NAME                                          CREATED AT
│ controlplanerevisions.mesh.cloud.google.com   2024-07-22T13:26:27Z
│ customresourcedefinition.apiextensions.k8s.io/controlplanerevisions.mesh.cloud.google.com
│ condition met
│ controlplanerevision.mesh.cloud.google.com/asm-managed-stable created
│ error: timed out waiting for the condition on
│ controlplanerevisions/asm-managed-stable
│ + cleanup
│ + rm -rf /tmp/kubectl_wrapper_875_8064
│ 
╵
Cleaning up project directory and file based variables

### Terraform Configuration

```hcl
`module "asm" {
  source                    = "terraform-google-modules/kubernetes-engine/google//modules/asm"
  project_id                = var.project_id
  cluster_name              = google_container_cluster.cluster.name
  cluster_location          = google_container_cluster.cluster.location
  enable_fleet_registration = true
}
```

### Terraform Version

```sh
1.9.3
```

**Comment by atul7799-dev** [comment]
below is the config  for ControlPlaneRevision

Name:         asm-managed-stable
Namespace:    istio-system
Labels:       app.kubernetes.io/created-by=terraform-module
              mesh.cloud.google.com/managed-cni-enabled=false
Annotations:  mesh.cloud.google.com/vpcsc: false
API Version:  mesh.cloud.google.com/v1beta1
Kind:         ControlPlaneRevision
Metadata:
  Creation Timestamp:  2024-07-28T20:05:23Z
  Generation:          1
  Managed Fields:
    API Version:  mesh.cloud.google.com/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:kubectl.kubernetes.io/last-applied-configuration:
          f:mesh.cloud.google.com/vpcsc:
        f:labels:
          .:
          f:app.kubernetes.io/created-by:
          f:mesh.cloud.google.com/managed-cni-enabled:
      f:spec:
        .:
        f:channel:
        f:type:
    Manager:      kubectl-client-side-apply
    Operation:    Update
    Time:         2024-07-28T20:05:23Z
    API Version:  mesh.cloud.google.com/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        .:
        f:conditions:
    Manager:         Google-GKEHub-Controllers-Servicemesh
    Operation:       Update
    Subresource:     status
    Time:            2024-07-28T20:05:34Z
  Resource Version:  1146760
  UID:               c7c81148-0c66-48a4-8b19-2e73971de96b
Spec:
  Channel:  stable
  Type:     managed_service
Status:
  Conditions:
    Last Transition Time:  2024-07-28T20:05:34Z
    Message:               The provisioning process has not completed successfully
    Reason:                NotProvisioned
    Status:                False
    Type:                  Reconciled
    Last Transition Time:  2024-07-28T20:05:34Z
    Message:               Provisioning has finished
    Reason:                ProvisioningFinished
    Status:                True
    Type:                  ProvisioningFinished
    Last Transition Time:  2024-07-28T20:05:34Z
    Message:               An internal error has occurred. Please contact customer support. This will be retried within 15 minutes.
    Reason:                PreconditionFailed
    Status:                True
    Type:                  Stalled
Events:                    <none>
